src/PrettyPrinter.ml

```diff
@@ -91,6 +118,16 @@ module Doc : DOC = struct
                ( letbox_kwd
                ^^^ !^(Id.M.to_string idm)
                ^^^ equals ^^^ walk 2 boxed ^^^ in_kwd ^/^ walk 1 body ))
+      | Match { matched; zbranch; pred; sbranch } ->
+          (parens_if (p > 1))
+            ( match_kwd ^^^ walk 1 matched ^^^ with_kwd ^/^ bar ^^^ zero_kwd
+            ^^^ darrow
+            ^^^ nest 10 (walk 1 zbranch)
+            ^/^ bar ^^^ succ_kwd
+            ^^^ !^(Id.R.to_string pred)
+            ^^^ darrow
+            ^^^ nest 10 (walk 1 sbranch)
```

Comment:
The level of nesting here should be computed from the lengths of the `succ` keyword, double arrow, whitespace, *and*  the length of the `pred` identifier, otherwise one gets something like this:
```
$ dune exec -- mtt parse -e "match 0 with | zero => 0 | succ n => match 1 with | zero => 1 | succ n => 1 end end"
match 0 with
| zero => 0
| succ n => match 1 with
          | zero => 1
          | succ n => 1
          end
end
```

Reply:
fixed